src/build.js

```diff
@@ -0,0 +1,43 @@
+const crypto = require("crypto")
+const url = require("url")
+
+class Build {
+  constructor(sqsMessage) {
+    this.sqsMessage = sqsMessage
+    this.buildID = this._generateBuildID()
+    this.containerEnvironment = this._resolveContainerEnvironment(
+      this.buildID,
+      this.sqsMessage
+    )
+  }
+
+  _buildCallbackURL(buildID) {
+    return url.resolve(
+      process.env.BUILD_COMPLETE_CALLBACK_HOST,
+      `builds/${buildID}/callback`
+    )
+  }
+
+  _generateBuildID() {
+    return crypto.randomBytes(48).toString('base64').
+                  replace(/\+/g, '-').
+                  replace(/\//g, '_').
+                  replace(/\=/g, '')
+  }
+
+  _resolveContainerEnvironment(buildID, sqsMessage) {
+    const containerOverrides = JSON.parse(sqsMessage.Body)
+    let environment = {}
+
+    for (let environmentOverride of containerOverrides.environment) {
```

Comment:
The data structure looks like this:

```json
{
  environment: [
    { name: "REPOSITORY", value: "owner/repo"
  ],
  name: "container-name"
}
```

You're right. Probably a good place to leverage `reduce.